### ISSUE TITLE
Stable serialization

### ISSUE THREAD

**Issue by Sytten** [juhaku/utoipa#210]
Since the library uses `HashMap`, the order of elements is not stable.
This means that the output is always different for each Yaml/Json serialization.
We use that yaml in code-reviews to make sure the API is conform to our standards and catch breaking changes, but this makes it impossible to do.

A possible solution would be to switch to a sorted map like a `BTreeMap` (breaking change since it's a public property).
Another solution would be to customize the serialization function to sort the keys.

**Comment by juhaku** [comment]
Yes some of the properties are using HashMap since I haven't thought it would be necessary to sort them always. But most of the items are using BTreeMap though. The items using HashMap could be changed to use BTreeMap quite easily though and change would be painless in next to next release since it will be breaking in anyways. I need to see whether this change is something that could be done to the next release. So far there hasn't been a requirement for everything to bee sorted the same always. But I understand the case where if the properties of the specific component changes it will be an issuea in terms of back wards compability check. So it would be easier to have them sorted always the same indeed.

**Comment by Sytten** [comment]
I am happy to contribute this change if you are ok with the BTreeMap usage. It should not be that complicated to do.

**Comment by juhaku** [comment]
Sure all contributions are welcome. And it is less work for me ;) Typically all new features need to be tested and documented, but since this is not a new feature nor it does change fundamentally existing outwarding API there nothing that actually need be tested or documented. As long as git commit message tells what changed and why. :)
